minikube.mk

```diff
@@ -0,0 +1,52 @@
+MINIKUBE_VERSION ?= v1.27.0
+KUBERNETES_VERSION ?= v1.24.6
+MINIKUBE_DRIVER ?= docker
+
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+MINIKUBE_BIN ?= $(PROJECT_DIR)/bin/minikube
+
+CONTAINER_RUNTIME ?= docker
+
+# make does not allow implicit rules (with '%') to be phony so let's use
+# the 'phony_explicit' dependency to make implicit rules inherit the phony
+# attribute
+.PHONY: phony_explicit
+phony_explicit:
+
+# minikube
+
+MINIKUBE = $(MINIKUBE_BIN)/minikube-$(MINIKUBE_VERSION)
+.PHONY: minikube-install
+minikube-install:
+	mkdir -p $(MINIKUBE_BIN)
+	test -e $(MINIKUBE_BIN)/minikube-$(MINIKUBE_VERSION) || \
+	(cd $(MINIKUBE_BIN) && curl -Lo ./minikube-$(MINIKUBE_VERSION) https://github.com/kubernetes/minikube/releases/download/$(MINIKUBE_VERSION)/minikube-linux-$(shell go env GOHOSTARCH))
```

Comment:
The main idea I like this is to allow including `minikube.mk` and not depend on anything from the including `Makefile`. I am using the idea [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/1241/files#diff-3041d09010d88ba810327d88ac25ddcd8d9753c32129f3b5a32f9fefab70eaf7R1). But it isn't a hard requirement I can adapt it if you think it will be better that way?